tests/panoptic/test_panoptic_pq.py

```diff
@@ -0,0 +1,197 @@
+from test_panoptic_utils import generate_synthetic_payload
+
+from seametrics.panoptic.pq import PanopticQuality
+from seametrics.panoptic.utils import payload_to_seg_metric
+
+
+def setup_panoptic_quality(area_range=None):
+    label2id = {
+        "WATER": 0,
+        "SKY": 1,
+        "LAND": 2,
+        "MOTORBOAT": 3,
+        "FAR_AWAY_OBJECT": 4,
+        "SAILING_BOAT_WITH_CLOSED_SAILS": 5,
+        "SHIP": 6,
+        "WATERCRAFT": 7,
+        "SPHERICAL_BUOY": 8,
+        "CONSTRUCTION": 9,
+        "FLOTSAM": 10,
+        "SAILING_BOAT_WITH_OPEN_SAILS": 11,
+        "CONTAINER": 12,
+        "PILLAR_BUOY": 13,
+        "AERIAL_ANIMAL": 14,
+        "HUMAN_IN_WATER": 15,
+        "OWN_BOAT": 16,
+        "WOODEN_LOG": 17,
+        "MARITIME_ANIMAL": 18,
+    }
+    stuff = ["WATER", "SKY", "LAND", "CONSTRUCTION", "ICE", "OWN_BOAT"]
+    per_class = True
+    split_sq_rq = True
+    area_rng = area_range if area_range is not None else [(0, 1e5**2), (0**2, 6**2), (6**2, 12**2), (12**2, 1e5**2)]
+    pq = PanopticQuality(
+        things=set(
+            [label2id[label] for label in label2id.keys() if label not in stuff]
+        ),
+        stuffs=set([label2id[label] for label in label2id.keys() if label in stuff]),
+        return_per_class=per_class,
+        return_sq_and_rq=split_sq_rq,
+        areas=area_rng,
+    )
+    return pq, label2id
+
+
+
+def test_panoptic_quality_initialization():
+    pq, _ = setup_panoptic_quality()
+    assert pq.device is not None, "Device should not be None"
+    assert pq.metric.return_per_class is True
+    assert pq.metric.return_sq_and_rq is True
+    assert len(pq.things) == 14
+    assert len(pq.stuffs) == 5
+    assert len(pq.things) + len(pq.stuffs) == 19
+    assert pq.CHUNK_SIZE == 200
+
+
+def test_panoptic_quality_areas():
+    pq, _ = setup_panoptic_quality()
+    assert pq.get_areas() == [
+        (0, 1e5**2),
+        (0**2, 6**2),
+        (6**2, 12**2),
+        (12**2, 1e5**2),
+    ], "Areas are not set correctly"
+
+
+def test_update_with_numpy_array():
+    pq, _ = setup_panoptic_quality()
+
+    preds = np.array([[[[3, 1], [4, 1]], [[7, 0], [8, 0]]]])
+    targets = np.array([[[[3, 1], [4, 1]], [[7, 0], [8, 0]]]])
+
+    pq.update(preds, targets)
+    assert pq.metric.true_positives.sum() > 0, "True positives should be updated"
+
+
+def test_update_with_torch_tensor():
+    pq, _ = setup_panoptic_quality()
+
+    preds = torch.tensor([[[[3, 1], [4, 1]], [[7, 0], [8, 0]]]])
+    targets = torch.tensor([[[[3, 1], [4, 1]], [[7, 0], [8, 0]]]])
+
+    pq.update(preds, targets)
+    assert pq.metric.true_positives.sum() > 0, "True positives should be updated"
+
+
+def test_compute_metric():
+    pq, label2id = setup_panoptic_quality()
+
+    payload = generate_synthetic_payload(gt_config=[1, 2], model_config=[1, 0])
+    pred, gt, label2id = payload_to_seg_metric(
+        payload=payload, model_name="model", label2id=label2id
+    )
+    pred[pred == -1] = 0
```

Comment:
If you like, you could add a comment here explaining why this is necessary, so that the code is easily understandable for everyone :)